File: Data Structures & Algorithms/partition-equal-subset-sum/submission-1.cpp
class Solution {
public:
    bool canPartition(vector<int>& nums) {
        int sum = accumulate(nums.begin(),nums.end(),0);
        if(sum%2 != 0){
            return false;
        }
        int target = sum/2;
        vector<bool> dp(target+1,false);
        dp[0]=true;
        for(int num: nums){
            for(int t=target; t>=num; t--){
                if(dp[t-num]){
                    dp[t]=true;
                }
            }
        }
        return dp[target];
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = dict()
        for i in range(len(nums)):
            comp = target - nums[i]
            if( comp in map ):
                if( i < map[comp]):
                    return [i,map[comp]]
                else:
                    return [map[comp],i]
            map[nums[i]]=i;
        return []